name: test Workflow

on:
    schedule:
      - cron: "0 */6 * * *"
    # push:
    #   branches: [ main ]
    
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
       - uses: actions/checkout@v2
       - name: Set up Python 3.9
         uses: actions/setup-python@v2.2.2
         with:
           python-version: 3.9

       - name: init
         continue-on-error: true
         run: |
           git config --global user.name ${{ secrets.NAME }}
           git config --global user.email ${{ secrets.MAIL }}
           git config --global credential.helper store
           echo "https://${{ secrets.NAME }}:${{ secrets.TOKEN }}@github.com" > ~/.git-credentials
       - name: cloning repo
         continue-on-error: true
         run: |
           git clone https://${{ secrets.NAME }}:${{ secrets.TOKEN }}@${{ secrets.REPO }} test
       - name: installing
         continue-on-error: true
         run: |
           cd test
           pip3 install -r requirements.txt
       - name: Running
         timeout-minutes: 350
         continue-on-error: true
         run: |
           cd test
           python3 bot.py
      #  - name: Loop workflow
      #    continue-on-error: true
      #    run: |
      #         git clone https://github.com/${GH_PATH} loop
      #         cd loop || exit 1
      #         echo $(date -u +%T%P) >> loop.txt
      #         git add loop.txt
      #         git commit -m "Workflow : Loop on $(date -u +%D-%T%P)"
      #         git push -q https://${{ secrets.TOKEN }}@github.com/${GH_PATH} HEAD:main
